### DISCUSSION TITLE
Parsing - keep comments

### DISCUSSION THREAD

**Discussion by PowalaX** [JSQLParser/JSqlParser#1721] [General]
Goal: SQL Rewrite with comments

Sample:
 SELECT NAME, ADDRESS, COL1 /* Important comment!*/ FROM **OLD_TABLE** WHERE SSN IN (**'1', '2'**);

After rewrite:
SELECT NAME, ADDRESS, COL1 /* Important comment!*/ FROM **NEW_TABLE** WHERE SSN IN (**'3', '4'**);

(SQL building with jSQLParser is clear.)
How can I use it with jSQLParser?

**Comment by manticore-projects** [comment]
Greetings.

Look at [JSQLFormatter](https://github.com/manticore-projects/jsqlformatter), where I create a map of the comments first based on its position considering only non-whitespace characters. Using that map you can re-insert the comments after de-parsing.
